src/components/library/library.jsx

```diff
@@ -172,33 +164,28 @@ class LibraryComponent extends React.Component {
                     })}
                     ref={this.setFilteredDataRef}
                 >
-                    {this.getFilteredData().map((dataItem, index) => {
```

Comment:
instead of determining which image url to use here, we pass the item's md5, its costumes object, and its rawURL down to the individual item and let it decide which to use